core/lib/src/config/secret_key.rs

```diff
@@ -178,6 +193,80 @@ impl SecretKey {
     {
         ser.serialize_bytes(&[0; 32][..])
     }
+
+    /// Encrypts the given data.
+    /// Generates a random nonce for each encryption to ensure uniqueness.
+    /// Returns the Vec<u8> of the concatenated nonce and ciphertext.
+    ///
+    /// # Example
+    /// ```rust
+    /// use rocket::config::SecretKey;
+    ///
+    /// let plaintext = "I like turtles".as_bytes();
+    /// let secret_key = SecretKey::generate().unwrap();
+    ///
+    /// let encrypted = secret_key.encrypt(&plaintext).unwrap();
+    /// let decrypted = secret_key.decrypt(&encrypted).unwrap();
+    ///
+    /// assert_eq!(decrypted, plaintext);
+    /// ```
+    pub fn encrypt<T: AsRef<[u8]>>(&self, value: T) -> Result<Vec<u8>, Error> {
+        // Convert the encryption key to a fixed-length array
+        let key: [u8; KEY_LEN] = self.key
+            .encryption()
+            .try_into()
+            .map_err(|_| Error::KeyLengthError)?;
+
+        // Create a new AES-256-GCM instance with the provided key
+        let aead = Aes256Gcm::new(GenericArray::from_slice(&key));
+
+        // Generate a random nonce
+        let mut nonce = [0u8; NONCE_LEN];
+        let mut rng = rand::thread_rng();
+        rng.try_fill_bytes(&mut nonce).map_err(|_| Error::NonceFillError)?;
+        let nonce = Nonce::from_slice(&nonce);
```

Comment:
Note that you can use the [`generate_nonce()`](https://docs.rs/aes-gcm/latest/aes_gcm/trait.AeadCore.html#method.generate_nonce) method to generate a properly sized nonce.

Reply:
Done!